File: src/contests/cg/puzzle/java/euclids-algorithm.java
// 🎮 CodinGame Puzzle - euclids-algorithm
// https://www.codingame.com/training/easy/euclids-algorithm

import java.io.*;

class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] p = br.readLine().trim().split(" ");
        long a = Long.parseLong(p[0]), b = Long.parseLong(p[1]);
        long x = a, y = b;
        while (y != 0) {
            long q = x / y, r = x % y;
            System.out.println(x + "=" + y + "*" + q + "+" + r);
            x = y;
            y = r;
        }
        System.out.println("GCD(" + a + "," + b + ")=" + x);
    }
}

File: src/contests/cg/puzzle/kotlin/euclids-algorithm.kt
// 🎮 CodinGame Puzzle - euclids-algorithm
// https://www.codingame.com/training/easy/euclids-algorithm

fun main() {
    val (a, b) = readLine()!!.trim().split(" ").map { it.toLong() }
    var x = a
    var y = b
    while (y != 0L) {
        val q = x / y
        val r = x % y
        println("$x=$y*$q+$r")
        x = y
        y = r
    }
    println("GCD($a,$b)=$x")
}
